### DISCUSSION TITLE
Sensor Fusion for Autonomous Systems (Project 233)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#67] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #233](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Sensor%20Fusion%20for%20Autonomous%20Systems)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by iletisiyorum** [comment]
hi. I need sensor or device list that compatible with Matlab. IMU and GPS which have driver

**Reply by robertogl** [reply]
Hello @iletisiyorum 

We still do not support a direct streaming from IMU sensors but you can use something like an Arduino board as explained [here](https://www.mathworks.com/help/fusion/ug/estimate-oientation-using-ahrs-filter-and-imu-data-in-simulink.html). An alternative could be getting IMU data from your phone using the MATLAB mobile app, although it might not be the best option.

Regarding GPS, you can connect a GPS directly to MATLAB and we do support that workflow with the [gpsdev object](https://www.mathworks.com/help/nav/ref/gpsdev-system-object.html).
There’s a list of GPS devices at the bottom of that ref page of devices that work with this API. Other GPSes may as well but haven’t been tested.

**Comment by iletisiyorum** [comment]
I am sorry to hearthat. Neither aurduino nor cell phone have scientific
backround, i think you have to fill the gap

Uğur Arslan
Flight Instructor


23 May 2022 Pzt 18:26 tarihinde robertogl ***@***.***> şunu
yazdı:

**Reply by robertogl** [reply]
Thanks for your feedback, Uğur!

**Comment by iletisiyorum** [comment]
You're welcome!

Uğur Arslan
Flight Instructor


13 Haz 2022 Pzt 23:52 tarihinde robertogl ***@***.***> şunu
yazdı:

**Comment by viv511** [comment]
Hi,

I am a rising high school junior at Lexington High School and am interested in working and collaborating on this project. It is my first time using MathWorks software so any tutorials or video links would be appreciated! :)

-viv511

**Reply by robertogl** [reply]
Hi @viv511,

Great to know you are interested in working on this project!

You can find plenty of tutorials, documentations, webinars, and online courses.

For example: 
- [MATLAB OnRamp](https://matlabacademy.mathworks.com/details/matlab-onramp/gettingstarted)
- [Get Started with MATLAB](https://www.mathworks.com/help/matlab/getting-started-with-matlab.html)
- [MathWorks Videos and Webinars](https://www.mathworks.com/videos/search.html?q=&page=1)
- [YouTube channel](https://www.youtube.com/user/MATLAB)

Moreover, detailed documentation is available on each toolbox page. Additionally, it may be helpful to have a basic understanding of Kalman filtering and sensors such as IMUs, GPS, and LIDARs. 

Best!

**Reply by viv511** [reply]
Thanks for the resources! I'll check them out soon. I've worked with many different sensors and have put together tracking systems for VEX robots, so I'm interested to see how that knowledge will carry over.

Regards, viv511

**Reply by viv511** [reply]
Hi Mr. Roberto,

Thank you once again for providing me with resources for this project. I would like an extension on the project submission deadline by around a few months. I was looking forward to working on this project, but due to travel and academic preparation, I don't think I can create with my best effort by the current deadline. 

Thanks for your consideration,
viv511

**Reply by robertogl** [reply]
Sure. That is totally fine. Would end of September work?

**Reply by viv511** [reply]
Yes that should work, thanks

**Reply by viv511** [reply]
I'm truly sorry, but I don't think I will be able to continue this project any further as of now. I may decide to come back at a later date to continue this, but I'm not sure when. Thank you for the amazing opportunity nevertheless.

**Reply by robertogl** [reply]
Ok no problem. You're welcome to continue with your project whenever you are ready or have time.

**Comment by saivignesh2** [comment]
sensor fusion for autonomous systems on arduino project  can you help on this project to do

**Reply by robertogl** [reply]
[Here](https://www.mathworks.com/help/fusion/ug/estimate-oientation-using-ahrs-filter-and-imu-data-in-simulink.html) an example of sensor fusion on Arduino.

**Reply by saivignesh2** [reply]
so that you are provided above is for autonomous systems linked or only sensor fusion
